cirq-core/cirq/ops/common_gates.py

```diff
@@ -316,6 +320,15 @@ def __repr__(self) -> str:
         all_args = ', '.join(args)
         return f'cirq.XPowGate({all_args})'
 
+    def _json_dict_(self) -> Dict[str, Any]:
+        d = protocols.obj_to_dict_helper(self, ['exponent', 'global_shift'])
+        if self.dimension != 2:
+            d['dimension'] = self.dimension
+        return d
+
+    def _value_equality_values_(self):
+        return self._canonical_exponent, self._global_shift, self._dimension
```

Comment:
I think it's better to call the `_value_equality_values_` of parent class and append it the tuple with `self.dimension`.  Also, we need to update approx equality as well (and add a test for the same). So, 

```suggestion
    def _value_equality_values_(self):
        return (*super()._value_equality_values_(), self._dimension)
        
    def _value_equality_approximate_values_(self):
        return (*super()._value_equality_approximate_values_(), self._dimension)
```

Ditto for ZPowGate.

Reply:
Should be pushed now. I also pushed up removing `test_json()` because I forgot to remove it in the previous round